Sage/carbon
# refactor(link): remove tabbable prop (FE-5089)

### Proposed behaviour

Remove redundant `tabbable` prop, update the documentation.

Render SubMenu parent item having `onClick` and no `href` as a button, by passing `onClick` to the Link component.

Remove redundant `href`, `target` and `rel` stories, as props described by them are just passed down and rendered as HTML attributes.

Group iconAlign and tooltip stories under the icon story, as these props are dependent on the icon being set.

Change `disabled` story description to match current behaviour.

### Current behaviour

The `tabbable` attribute is redundant, as the component renders either anchor or a button, in which tabbable property could be toggled by disabling a button or not specifying a href in an anchor.

When the Menu is displayed in fullscreen mode, SubMenu parent item with `onClick` prop passed and no `href` is rendered as link instead of a button.

### Checklist



~~- [ ] Related issues linked in commit messages if required~~
~~- [ ] Screenshots are included in the PR if useful~~
~~- [ ] All themes are supported if required~~
~~- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required~~

#### QA


### Additional context

fixes #5270

### Testing instructions

Components affected by the change: Link, Menu, TextEditor (editor link).

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.
